software/contrib/piconacci.py

```diff
@@ -0,0 +1,145 @@
+# trigger every 300 years which is quite a lot even for slow moving
+# ambient pieces,
+MAX_FIB = 50
+
+
+class Piconacci(EuroPiScript):
+    def __init__(self):
+        # Flag to indicate whether the display needs updating.
+        self.dirty = True
+        # offset determines the sublist to be used for the values.
+        self.offset = 0
+        # rotate adjusts how the values are mapped to the cv
+        # outputs. A value of 0 means that cv1 is controlled by the
+        # first, cv2 the second etc. A rotate value of 1 means that
+        # cv2 is controlled by the first, cv3 the second, and cv1 the
+        # last.
+        self.rotate = 0
+        # Overclock the Pico for improved performance.
+        machine.freq(250_000_000)
+        # List of fibonacci numbers.
+        self.fib = [1, 1]
+        for i in range(0, MAX_FIB):
+            self.fib.append(self.fib[-2] + self.fib[-1])
+        # Strip off first 1
+        self.fib = self.fib[1:]
+
+        self.steps = [0] * 6
+
+        # Triggered when button 1 is released
+        # Short press:
+        # Long press:
+        @b1.handler_falling
+        def b1Pressed():
+            if ticks_diff(ticks_ms(), b1.last_pressed()) > 300:
+                # Rotate values left
+                self.rotate = (self.rotate - 1) % 6
+            else:
+                # Shift all the values left in the sequence
+                if self.offset > 0:
+                    self.offset -= 1
+            # State has changed
+            self.dirty = True
+
+        # Triggered when button 2 is released.
+        # Short press:
+        # Long press:
+        @b2.handler_falling
+        def b2Pressed():
+            if ticks_diff(ticks_ms(), b2.last_pressed()) > 300:
+                # Rotate values right
+                self.rotate = (self.rotate + 1) % 6
+            else:
+                # Shift all the values right in the sequence
+                if self.offset < MAX_FIB - 7:
+                    self.offset += 1
+            # State has changed
+            self.dirty = True
+
+        # Triggered on each clock into digital input. Output triggers.
+        @din.handler
+        def clockTrigger():
+            for tr in range(0, 6):
+                self.steps[tr] += 1
+                if self.steps[tr] >= self.value(tr):
+                    # Trigger on tr
+                    cvs[tr].on()
+                    # If the values change due to shift or rotation,
+                    # we may get slightly odd behaviour here,
+                    # particularly if the values reduce.
+                    self.steps[tr] = 0
+
+        @din.handler_falling
+        def clockTriggerEnd():
+            for cv in cvs:
+                cv.off()
+
+    def value(self, index):
+        values = self.fib[self.offset : self.offset + 6]
+        return values[(index + self.rotate) % 6]
+
+    def main(self):
+        # Reset module display state.
+        oled.fill(0)
```

Comment:
[optional] I don't think that this is necessary here. The initial state is `dirty = True`, so the first trip through the while loop below will update the screen.